File: NERP.Admin.Web1/ClientApp/src/app/entities/models/homepass.ts
import { Point } from '../../common/geo/geometry';
import { CustomField } from '../../common/models/customField';
import { DbBaseEntity } from '../../common/models/dbBaseEntity';
import { FileAttachment } from '../../common/models/FileAttachment';

export class Homepass extends DbBaseEntity {
    name: string;
    icon?: string;
    zoomLevel?: number;

    code: string;
    description?: string;
    type?: string;
    geometry: Point;

    customFields?: CustomField[] = new Array<CustomField>();
    photos?: FileAttachment[] = new Array<FileAttachment>();
    attachments?: FileAttachment[] = new Array<FileAttachment>();

}

File: NERP.Homepasses/Models/Homepass.cs
using Common.DataAccess.Mongo;
using NERP.Common.Models;
using System.Collections.Generic;

namespace NERP.Homepasses.Model
{
    public class Homepass : DbBaseEntity
    {
        public string Name { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }
        public string Type { get; set; }
        public string Status { get; set; }
        public Point Geometry { get; set; }
        public Polygon Area { get; set; }

        public int ZoomLevel { get; set; }
        public List<CustomField> CustomFields { get; set; }
        public List<FileAttachment> Photos { get; set; }
        public List<FileAttachment> Attachments { get; set; }

        public int Connections { get; set; }
    }
}
